spec/controllers/books_controller_spec.rb

```diff
@@ -0,0 +1,89 @@
+require 'rails_helper'
+
+describe BooksController do
+  let!(:books) { create_list(:book, 3) }
+
+  describe 'GET #index' do
+    context 'With an unauthenticated user' do
+      it 'returns an unauthorized status' do
+        get :index, params: {}
+        expect(response).to have_http_status(:unauthorized)
+      end
+    end
+
+    context 'With an authenticated user' do
+      include_context 'Authenticated User'
+
+      it 'without params returns all books' do
+        get :index
+        expect(JSON.parse(response.body)['total_count']).to eq(books.size)
+      end
+
+      it 'returns a 200 ok status' do
+        get :index, params: { book: { genre: books.first.genre } }
+        expect(response).to have_http_status(:ok)
+      end
+
+      def unique_title
```

Comment:
It isn't a common practice declare `def`s in specs files, we usually do it to encapsulate some complicated logic necessary to make an expectation. I don't think this is that case.